lib/Frontend/Frontend.cpp

```diff
@@ -450,6 +450,16 @@ bool CompilerInstance::setUpModuleLoaders() {
     return true;
   }
 
+  // If implicit modules are disabled, we need to install an explicit module
+  // loader.
+  if (Invocation.getFrontendOptions().DisableImplicitModules) {
```

Comment:
How about modeling `DisableImplicitModules` as a `ModuleLoadingMode`? It fits in with the scheme established there.